ruby/ql/lib/codeql/ruby/ast/internal/TreeSitter.qll

```diff
@@ -48,6 +53,38 @@ module Ruby {
     final override string getAPrimaryQlClass() { result = "ReservedWord" }
   }
 
+  /** Gets the file path of the given `node`. */
+  overlay[local]
+  pragma[nomagic]
+  string getRawFile(@ruby_ast_node node) {
```

Comment:
IIRC the name `getRawFile` comes from the Java discard predicates and makes sense there because Java has a translation layer in QL where we wanted to get file names on the raw/input side of it. It doesn't necessarily make sense for Ruby.